app/main.py

```diff
@@ -1,3 +1,67 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: int) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        elif isinstance(other, (int, float)):
+            return Distance(
+                km=self.km + other
+            )
+
+    def __iadd__(self, other: int) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        elif isinstance(other, (int, float)):
+            self.km += other
+            return self
+
+    def __mul__(self, other: int) -> "Distance":
```

Comment:
The `__mul__` method should return a new `Distance` instance instead of modifying the current instance. Consider using `return Distance(self.km * other)`.